plugin/pkg/admission/resourcequota/controller.go

```diff
@@ -506,7 +508,14 @@ func CheckRequest(quotas []corev1.ResourceQuota, a admission.Attributes, evaluat
 			if innerErr != nil {
 				return quotas, innerErr
 			}
-			deltaUsage = quota.SubtractWithNonNegativeResult(deltaUsage, prevUsage)
+			if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+				// allow negative usage for pods as pod resources can increase or decrease
+				if a.GetResource().GroupResource() == corev1.Resource("pods") {
+					deltaUsage = quota.Subtract(deltaUsage, prevUsage)
+				}
```

Comment:
this leaves non-pod resources with deltaUsage uncomputed... need a test case that would have caught this

Reply:
remind me what protects against making requests that 1) reduce pod resource usage and 2) are constructed to fail to persist (e.g. a conflicting resourceVersion) and artificially lowering consumed quota?

Reply:
Subtract - 🤦‍♂️  .. will fix and add test.

Regarding gaming of resource quota, we decided against using max(Resources.Requests, ResourcesAllocated) as we don't realistically expect a downward resize to stay unenforced by the kubelet for an extended period of time. Kubelet will always accept downward resize typically in the watch latency.

Please review the rationale noted on the KEP discussion for requests: 
- [resource quota without max](https://github.com/kubernetes/enhancements/pull/686/#discussion_r318246172)()
- [kubelet handling resize without max](https://github.com/kubernetes/enhancements/pull/686/#discussion_r318246172)()

Similarly for limits, iirc [we expected CRI to converge](https://github.com/kubernetes/enhancements/pull/686/#discussion_r324806067) to the new desired limits shortly after watch latency.